pkg/plugin/aggregation/update.go

```diff
@@ -166,25 +174,41 @@ func GetPatch(annotation string) map[string]interface{} {
 	}
 }
 
-// GetStatusPodName gets the sonobuoy master pod name based on its label.
-func GetStatusPodName(client kubernetes.Interface, namespace string) (string, error) {
+// GetAggregatorPod gets the sonobuoy aggregator pod based on its label.
+func GetAggregatorPod(client kubernetes.Interface, namespace string) (*v1.Pod, error) {
 	listOptions := metav1.ListOptions{
 		LabelSelector: StatusPodLabel,
 	}
 
 	podList, err := client.CoreV1().Pods(namespace).List(listOptions)
 	if err != nil {
-		return "", errors.Wrap(err, "unable to list pods with label %q")
+		return nil, errors.Wrap(err, "unable to list pods with label %q")
 	}
 
 	switch {
 	case len(podList.Items) == 0:
-		logrus.Warningf("No pods found with label %q in namespace %s, using default pod name %q", StatusPodLabel, namespace, DefaultStatusPodName)
-		return DefaultStatusPodName, nil
+		return nil, NoPodWithLabelError(fmt.Sprintf("no pods found with label %q in namespace %s", StatusPodLabel, namespace))
+
 	case len(podList.Items) > 1:
-		logrus.Warningf("Found more than one pod with label %q. Using %q", StatusPodLabel, podList.Items[0].GetName())
-		return podList.Items[0].GetName(), nil
+		logrus.Warningf("Found more than one pod with label %q. Using pod with name %q", StatusPodLabel, podList.Items[0].GetName())
+		return &podList.Items[0], nil
 	default:
-		return podList.Items[0].GetName(), nil
+		return &podList.Items[0], nil
+	}
+}
+
+// GetAggregatorPodName gets the sonobuoy aggregator pod name based on its label.
+func GetAggregatorPodName(client kubernetes.Interface, namespace string) (string, error) {
+	ap, err := GetAggregatorPod(client, namespace)
+
+	if _, ok := err.(NoPodWithLabelError); ok {
```

Comment:
nit: when doing multi-step error checking like this I tend to prefer a switch statement so it is a single block and the `if err != nil` section doesn't drift away from the call generating the error.

```
if err != nil {
        switch err.(type) {
        case * NoPodWithLabelError:
            fmt.Println(err.Error())
        default:
            fmt.Println("ah!?")
        }
    }
```

It is one extra layer of indentation, but effectively one unit of error handling. (again, just a nit)